src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -202,6 +215,15 @@ public ZooKeeperSaslClient getSaslClient() {
 
     private final ZKWatchManager watchManager;
 
+    private ZKClientConfig clientConfig;
+
+    public ZKClientConfig getClientConfig() {
```

Comment:
+1, about moving the instantiation block to constructor e making ``clientConfig`` final. :+1::